libr/asm/p/asm_mips_gnu.c

```diff
@@ -43,8 +45,38 @@ static int disassemble(struct r_asm_t *a, struct r_asm_op_t *op, const ut8 *buf,
 	Offset = a->pc;
 	memcpy (bytes, buf, 4); // TODO handle thumb
 
+	if ((a->cpu != pre_cpu) && (a->features != pre_features)) {
+		free (disasm_obj.disassembler_options);
+		memset (&disasm_obj, '\0', sizeof (struct disassemble_info));
+	}	
+
 	/* prepare disassembler */
-	memset (&disasm_obj,'\0', sizeof (struct disassemble_info));
+	if (a->cpu && a->cpu != pre_cpu) {
+		if (!r_str_casecmp (a->cpu, "mips64r2")) {
+			disasm_obj.mach = bfd_mach_mipsisa64r2;
+		} else if (!r_str_casecmp (a->cpu, "mips32r2")) {
+			disasm_obj.mach = bfd_mach_mipsisa32r2;
+		} else if (!r_str_casecmp (a->cpu, "mips64")) {
+			disasm_obj.mach = bfd_mach_mipsisa64;
+		} else if (!r_str_casecmp (a->cpu, "mips32")) {
+			disasm_obj.mach = bfd_mach_mipsisa32;
+		}
+		pre_cpu = a->cpu;
+	}
+
+	if (a->features && a->features != pre_features) {
+		char *ops = NULL;
```

Comment:
i dont see why are you appending a string that is always empty and can be concatenated only once. why not just strdup on each if case? because theres no way to have more than one abi set.

also, the trailing comma should be there. usually if i have to concat i use one of the following techniques:

* push elements in RList, and then join with ","
* concat a string with an optional comma before the word that depends on the string to be empty or not

Reply:
Breakpoint 3, disassemble () at p/asm_mips_gnu.c:67
(gdb) p	/s a->features 
$3 = **0x5555558376a0** "n64"
(gdb) continue 
Continuing.

Breakpoint 3, disassemble () at p/asm_mips_gnu.c:67
(gdb) p /s a->features 
$4 = **0x5555558376a0** "o32"
(gdb) 

I found that pre_features did not play the expected role, we decided to save some strstr() spent unnecessary time

Reply:
I should delete "if (a->features && a->features != pre_features)"